net/socket/tcp_socket_starboard.cc

```diff
@@ -50,7 +55,32 @@ TCPSocketStarboard::~TCPSocketStarboard() {
 int TCPSocketStarboard::Open(AddressFamily family) {
   DCHECK_CALLED_ON_VALID_THREAD(thread_checker_);
   DCHECK(!SbSocketIsValid(socket_));
+#if SB_API_VERSION < 16
   socket_ = SbSocketCreate(ConvertAddressFamily(family), kSbSocketProtocolTcp);
+#else
+  int socket_fd = socket(ConvertSocketAddressTypeToDomain(ConvertAddressFamily(family)), 
+                    SOCK_STREAM, IPPROTO_TCP);
+  if (socket_fd < 0) {
+    socket_= kSbSocketInvalid;
+  } else if (!SocketSetNonBlocking(socket_fd)) {
+    // Something went wrong, we'll clean up (preserving errno) and return
+    // failure.
+    int save_errno = errno;
+    HANDLE_EINTR_WRAPPER(close(socket_fd));
+    errno = save_errno;
+    socket_ = kSbSocketInvalid;
+  } else {
+    socket_ = new SbSocketWrapper(ConvertAddressFamily(family), kSbSocketProtocolTcp, socket_fd);
```

Comment:
Why do you need a SbSocketWrapper?

Reply:
Structure SbSocket members are associated to each other, like the address, the type, the error, and socket_waiter. It is convenient to keep them in a group.